query/src/servers/http/middleware.rs

```diff
@@ -38,14 +39,22 @@ pub struct HTTPSessionMiddleware {
     pub session_manager: Arc<SessionManager>,
 }
 
-fn get_credential(headers: &HeaderMap) -> Result<Option<Credential>> {
+fn get_credential(headers: &HeaderMap) -> Result<Credential> {
     let auth_headers: Vec<_> = headers.get_all(AUTHORIZATION).iter().collect();
     if auth_headers.len() > 1 {
         let msg = &format!("Multiple {} headers detected", AUTHORIZATION);
         return Err(ErrorCode::AuthenticateFailure(msg));
     }
     if auth_headers.is_empty() {
-        return Ok(None);
+        return Err(ErrorCode::AuthenticateFailure(
+            "No authorization header detected",
+        ));
+    }
+    let mut hostname = headers.get(HOST).unwrap().to_str().unwrap_or("%");
```

Comment:
Do we need to handle the unwrap ?

Reply:
I changed to use `remote_addr` of request.